Zscaler/zscaler/block_ioc.py

```diff
@@ -0,0 +1,126 @@
+from collections import defaultdict
+from typing import Dict
+from zscaler.helpers import stix_to_indicators
+
+from sekoia_automation.action import Action
+from zscaler_api_talkers import ZiaTalker
+from requests.exceptions import JSONDecodeError
+
+
+class ZscalerAction(Action):
+    def zia_auth(self):
+        try:
+            api = ZiaTalker(self.module.configuration["base_url"])
+            api.authenticate(
+                api_key=self.module.configuration["api_key"],
+                username=self.module.configuration["username"],
+                password=self.module.configuration["password"],
+            )
+            return api
+        except Exception as e:
+            print(f"ZIA authentication failed: {str(e)}")
+            return None
+
+    def get_valid_indicators_from_list(self, arguments) -> list:
+        try:
+            IOC_list = [arguments["IoC"]]
+            self.log(f"IOC_list to block {IOC_list}")
+            return IOC_list
+        except Exception as e:
+            print(f"Build of IOC list failed: {str(e)}")
+            return []
+
+    def get_valid_indicators_from_stix(self, stix_objects):
+        """
+        Transform the IOC from the STIX format to the payload expected by Zscaler
+        """
+        ZSCALER_IOC_TYPE: Dict[str, Dict[str, str]] = {
+            "ipv4-addr": {"value": "ipv4"},
+            "ipv6-addr": {"value": "ipv6"},
+            "domain-name": {"value": "domain"},
+            "url": {"value": "url"},
+        }
+        seen_values = defaultdict(list)
+        results = {"valid": [], "revoked": []}
+        try:
+            for object in stix_objects:
+                # Extract value and type from pattern
+                print(f"object {object}")
+                indicators = stix_to_indicators(stix_object=object, supported_types_map=ZSCALER_IOC_TYPE)
+                print(f"indicators {indicators}")
+                for indicator in indicators:
+                    ioc_value = indicator["value"]
+                    ioc_type = indicator["type"]
+                    print(f"ioc_value {ioc_value}, ioc_type {ioc_type}")
+                    if ioc_type in ZSCALER_IOC_TYPE:
+                        # Handle revoked objects
+                        if object.get("revoked", False):
+                            results["revoked"].append(ioc_value)
+                            continue
+
+                        # Ignore this IOC if the same value has already been processed
+                        if ioc_value in seen_values[ioc_type]:
+                            continue
+
+                        seen_values[ioc_type].append(ioc_value)
+                        results["valid"].append(ioc_value)
+        except Exception as e:
+            print(f"Build of IOC list failed: {str(e)}")
```

Comment:
Use [`self.log_exception`](https://github.com/SEKOIA-IO/sekoia-automation-sdk/blob/main/sekoia_automation/module.py#L395) to display errors.
```suggestion
            self.log_exception(e, message=f"Build of IOC list failed")
```